CHANGELOG.md

```diff
@@ -13,15 +13,30 @@ All notable changes to this library are documented in this file.
 - BFV: replaced `bfv.Evaluator.InnerSum` with the more complete `rlwe.Evaluator.InnerSum`.
 - BFV: the `Evaluator` addition and subtraction no longer enforce BFV-specific operand types.
 - BFV: the maximum degree allowed for ciphertext multiplication has been reduced to two (same as `bgv` and `ckks`).
+- BFV: removed checks during addition and subtraction for the type of plaintext.
 - CKKS: added the `Polynomial.Lazy` field which specifies if the power basis is computed with lazy relinearization.
 - CKKS: made `NttAndMontgomery` thread safe again!
 - CKKS: removed `Evaluator` methods `MultByGaussianInteger`, `MultByGaussianIntegerThenAdd`, `MultByi`, `MultByiNew`, `DivByi` and `DivByiNew`. These are now all handled by the methods `MultByConst[...]`.
 - CKKS: updated the behavior of `MultByConstAndAdd`.
 - CKKS: fixed the median statistics of `PrecisionStats`, that were off by one index. 
+- CKKS: renamed the `Parameters` field `DefaultScale` to `LogScale`, which now takes a value in log2.
+- CKKS: the `Parameters` field `LogSlots` now has a default value which is the maximum number of slots possible for the given parameters.
+- CKKS: variable `BSGSRatio` is now `LogBSGSRatio` and is given in log2.
+- CKKS/Bootstrapping: complete refactoring the bootstrapping parameters for better usability.
+- CKKS/Bootstrapping: upon bootstrapping, the method will check that the ciphertext scale is a power of two.
+- CKKS/Bootstrapping: added the iterative bootstrapping `META-BTS` of `Youngjin et al.`.
+- CKKS/Bootstrapping: added `SimpleBootstrapper` which provides a re-encryption using the secret key and complies to the `rlwe.Bootstrapper` interface.
+- CKKS/Advanced: refactored names of structs and methods of the homomorphic encoding/decoding to better convey they purpose.
+- CKKS/Advanced: all fields of `EncodingMatrixLiteral` are now marshalled.
+- CKKS/Advanced: the `CoeffsToSlots` matrix is only scaled by an additional factor 1/2 if the `RepackImag2Real` field is set to true.
+- DCKKS: `GetMinimumLevelForBootstrapping` has been renamed to `GetMinimumLevelForRefresh`.
 - RLWE: added `CheckBinary` and `CheckUnary` to the `Evaluator` type. It performs pre-checks on operands of the `Evaluator` methods.
-- RLWE: added the `MaxLevelQ` and `MaxLevelP` methods to the `Parameters` type.
+- RLWE: added the methods `MaxLevelQ()` and `MaxLevelP` to the `Parameters` struct.
 - RLWE: added the method `NewCiphertextQP`.
-- RLWE: setting the Hamming weight of the secret or the standard deviation of the error to negative values will now instantiate these fields as zero values and return a warning.
+- RLWE: setting the Hamming weight of the secret or the standard deviation of the error to negative values will instantiate these fields as zero values and return a warning (as an error).
```

Comment:
Make it clearer that this is for the `NewParameter` only (the same behavior for `NewParameterFromLiteral` would be far too error-prone).

Reply:
done